### DISCUSSION TITLE
Erroneous (?) Binary Flags in the X-ray Flux Datasets

### DISCUSSION THREAD

**Discussion by RomainHautier** [ARCLab-MIT/STORM-AI-devkit-2025#6] [Q&A] [answered]
Hello,

After exploring the X-ray flux datasets (both GOES-13 X-Ray Flux Data sets) I noticed that variables which are supposed to be binary flags (**xrsa_flag**, **xrsb_flag**, **xrsa_flag_excluded**, **xrsb_flag_excluded**) take a variety of integer values ranging from 0 to 256. 
I would have thought binary flags to be either 0s or 1s, helping us judge the validity of the data at hand. 

Would you be able to clarify the confusion please?

I appreciate any help you can provide.

Regards,

Romain.

**Comment by hlizsolera** [comment]
Hi Romain,

After reviewing the GOES data, I can see that the `xrsa_flag` and `xrsb_flag` are always `16` when they should be `0` or `17` when they should be `1`. I can also see many different integer values for the other two flags as you mentioned. 

Thank you for bringing this to our attention and for your patience while waiting for an update. Our response time is longer this week since several members of our team are away at conferences or finishing final exams, but we are looking into this and will share our findings as soon as possible.

Best Regards,
Liz

**Comment by RomainHautier** [comment]
Hello Liz,

Thank you for getting back to me. No worries, I'll wait patiently for more
precision regarding this matter.

Have a good day and a nice weekend,

Regards,

Romain.

Le jeu. 12 déc. 2024 à 20:28, Liz Solera ***@***.***> a
écrit :

**Reply by edennachen** [reply, marked as answer]
Hi Romain, 

Thanks for your question and for your patience! After review, this was a misinterpretation of the data on our part. We sincerely apologize and will be updating the relevant section of the data description page. 

**Overall, it would be most helpful for you to just think of any data entries with 0.0 flag values as ‘good’ data. Any other value in the flag columns can be considered ‘bad’ or ‘compromised’ data.**

Here is the updated description: 

For earlier GOES satellite data (up until about 2017), you can assume the following bit arrangement for xrsa_flag and xrsb_flag values:
      0 good_data
      1 eclipsed_by_earth
      2 saturation
      3 bad_missing_data
      4 temperature_recovery

For later GOES satellite data (past 2017), xrsa_flag and xrsb_flag values take the following bit arrangement:
      0  good_data
      1  bad_data
      2  eclipse_by_earth
      3  temperature_recovery
      4  electron_correction_valid
      5  electron_correction_invalid
      6  electron_correction_interp
      7  electron_correction_decay

There are more GOES satellite data caveats you might find helpful [at this link](https://www.noaasis.noaa.gov/pdf/ps-pvr/goes-16/EXIS/XRS/Full/G16_GOES-R_XRS_L1b_Operational_Data_Readme.pdf), or you could go more into the weeds [the ancillary data and metadata of the original data files](https://www.ospo.noaa.gov/resources/documents/PUG/GS%20Series%20416-R-PUG-Main-0345%20Vol%201%20Rev%202.3.pdf) and [the flag conventions](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/ch03s05.html).

Let us know if you have any follow-up questions.
